save-backend/src/main/kotlin/com/saveourtool/save/backend/service/TestsSourceVersionService.kt

```diff
@@ -81,22 +74,30 @@ class TestsSourceVersionService(
         organizationName: String,
         sourceName: String,
     ): List<TestsSourceVersionInfo> =
-            versionRepository.findAllBySnapshot_Source_Organization_NameAndSnapshot_Source_Name(organizationName, sourceName)
+            versionRepository.findAllBySnapshot_Source_OrganizationNameAndSnapshot_SourceName(organizationName, sourceName)
                 .map(TestsSourceVersion::toInfo)
 
     /**
      * @param organizationName
      * @param sourceName
-     * @return all fetched version of [TestsSourceSnapshot] found by provided values
+     * @return all fetched version of [TestsSourceVersion] found by provided values
      */
     fun getAllVersions(
         organizationName: String,
         sourceName: String,
-    ): Set<String> = versionRepository.findAllBySnapshot_Source_Organization_NameAndSnapshot_Source_Name(
+    ): Set<String> = versionRepository.findAllBySnapshot_Source_OrganizationNameAndSnapshot_SourceName(
         organizationName,
         sourceName
     ).mapTo(HashSet(), TestsSourceVersion::name)
 
+    /**
+     * @param snapshotId
+     * @return all fetched version of [TestsSourceVersion] found by provided values
+     */
+    fun getAllVersions(
+        snapshotId: Long,
+    ): Set<String> = versionRepository.findAllBySnapshotId(snapshotId).mapTo(HashSet(), TestsSourceVersion::name)
```

Comment:
May I suggest that you replace a HS with an LHS here, so that the iteration order of versions is preserved?

Reply:
changed